x-pack/plugins/ml/server/plugin.ts

```diff
@@ -74,11 +76,14 @@ export class MlServerPlugin
   private security: SecurityPluginSetup | undefined;
   private isMlReady: Promise<void>;
   private setMlReady: () => void = () => {};
+  private readonly kibanaIndexConfig: Observable<{ kibana: { index: string } }>;
 
   constructor(ctx: PluginInitializerContext) {
     this.log = ctx.logger.get();
     this.mlLicense = new MlLicense();
     this.isMlReady = new Promise((resolve) => (this.setMlReady = resolve));
+
+    this.kibanaIndexConfig = ctx.config.legacy.globalConfig$;
```

Comment:
You can use `ctx.config.legacy.get()` instead, which is the synchronous counterpart.

Reply:
Updated in https://github.com/elastic/kibana/pull/97727/commits/a25b6403b7cb0e780a759fad0d60d902c6a089d5